projects/components/src/utils/test/widget-object/cypress-widget-finder.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Type } from '@angular/core';
+import { IdGenerator } from '../../id-generator/id-generator';
+import { Chainable, CypressLocatorDriver } from './cypress-widget-object';
+import { BaseWidgetObject, CorrectReturnTypes, TaggedClass } from './widget-object';
+
+const idGenerator = new IdGenerator('cy-id');
+
+/**
+ * Knows how to find Cypress widget objects within the DOM.
+ */
+export class CypressWidgetObjectFinder {
+    /**
+     * Finds a single widget object
+     *
+     * @param widgetConstructor the constructor of the widget to use
+     * @param ancestor the parent to begin the search from
+     * @param cssSelector the cssSelector to post-pend to the tagName for the search
+     *
+     */
+    public find<T extends TaggedClass & Type<BaseWidgetObject<Chainable>>>(
```

Comment:
I am wondering if it is necessary to create new CypressWidgetObjectFinder object every time using find function like `new CypressWidgetObjectFinder`. Since this class only has this function, it seems like a static function or export function is enough to find widget object.

Reply:
I tried to implement this, and while in Cypress it might make sense to have this just be a function, the AngularWidgetFinder exposes useful information in tests. Namely, it's what allows the user to run detectChanges and get access to the component if needed (like in the data-exporter tests). Furthermore, the Angular finder also creates the component in its constructor if given a component type, and this conceptually would be weird to move to the `find` method. 

So I guess we could make the Cypress finder a function and keep the Angular finder a class, but that feels unnecessary. For now, I'm going to keep the finder's unchanged as far as this comment goes, but let me know what you think.

Reply:
Yes, since angular finder class is more complex, I think we can keep this for consistency